project_cost_report/report/report.py

```diff
@@ -0,0 +1,320 @@
+        time_total_hours = self._get_time_total_hours(project)
+        time_sale_price = self._get_total_section_sale_price(time_categories)
+        outsourcing_categories = self._get_outsourcing_categories(
+            project, report_context
+        )
+        outsourcing_total = self._get_outsourcing_total(project)
+        outsourcing_sale_price = self._get_total_section_sale_price(
+            outsourcing_categories
+        )
+
+        total_cost = product_total + time_total + outsourcing_total
+        total_sale_price = product_sale_price + time_sale_price + outsourcing_sale_price
+        total_profit = total_sale_price - total_cost
+        total_margin = (
+            (total_profit / total_sale_price) * 100 if total_sale_price else 0
+        )
+
+        waiting_purchase_orders = self._get_waiting_purchase_orders(project)
+        waiting_purchase_orders_total = sum(
+            get_waiting_for_invoice_total(o, project) for o in waiting_purchase_orders
+        )
+
+        return {
+            "adjust_analytic_line_amount_sign": adjust_analytic_line_amount_sign,
+            "currency": project.company_id.currency_id,
+            "is_outsourcing_line": is_outsourcing_line,
+            "is_product_line": is_product_line,
+            "is_timesheet_line": is_timesheet_line,
+            "outsourcing_categories": outsourcing_categories,
+            "outsourcing_total": outsourcing_total,
+            "print_date": print_date,
+            "product_categories": product_categories,
+            "product_total": product_total,
+            "project": project,
+            "show_summary": report_context.get("show_summary", True),
+            "time_categories": time_categories,
+            "time_total": time_total,
+            "time_total_hours": time_total_hours,
+            "total_cost": float_round(total_cost, 2),
+            "total_profit": float_round(total_profit, 2),
+            "total_target_margin": float_round(total_margin, 2),
+            "total_target_sale_price": float_round(total_sale_price, 2),
+            "waiting_purchase_orders": waiting_purchase_orders,
+            "waiting_purchase_order_total": float_round(
+                waiting_purchase_orders_total, 2
+            ),
+            "get_waiting_for_invoice_total": get_waiting_for_invoice_total,
+        }
+
+    def get_project_from_report_context(self, report_context):
+        """Get the project record from the given report context.
+
+        :param report_context: the rendering context
+        :return: a project.project singleton
+        """
+        project_id = report_context.get("active_id")
+
+        if not project_id:
+            raise ValidationError(
+                _("The cost report was triggered without a project ID in context.")
+            )
+
+        return self.env["project.project"].browse(project_id)
+
+    @api.model
+    def get_html(self, report_context):
+        """Get the report html given the report context.
```

Comment:
Codacy found an issue: [Multi-line docstring summary should start at the second line (D213)](https://app.codacy.com/gh/Numigi/odoo-project-addons/pullRequest?prid=6165869)